shell.nix

```diff
@@ -0,0 +1,53 @@
+{ pkgs ? import <nixpkgs> {} }:
+	let
+	overrides = (builtins.fromTOML (builtins.readFile ./rust-toolchain.toml));
+	libPath = with pkgs; lib.makeLibraryPath [
+		# load external libraries that you need in your rust project here
+	];
+in
+pkgs.mkShell rec {
+	# Kanidm dependencies
+	buildInputs = with pkgs; [
+		pkg-config
+		
+		git
+		rustup
+
+		clang
+		llvmPackages.bintools
+		
+		openssl
+		systemd
+		linux-pam
+	];
+	
+	RUSTC_VERSION = overrides.toolchain.channel;
+	# https://github.com/rust-lang/rust-bindgen#environment-variables
+	LIBCLANG_PATH = pkgs.lib.makeLibraryPath [ pkgs.llvmPackages_latest.libclang.lib ];
+	shellHook = ''
+		export PATH=$PATH:''${CARGO_HOME:-~/.cargo}/bin
+		export PATH=$PATH:''${RUSTUP_HOME:-~/.rustup}/toolchains/$RUSTC_VERSION-x86_64-unknown-linux-gnu/bin/
+		'';
+
+	# Add precompiled library to rustc search path
+	RUSTFLAGS = (builtins.map (a: ''-L ${a}/lib'') [
+		# add libraries here (e.g. pkgs.libvmi)
+	]);
+	LD_LIBRARY_PATH = libPath;
+	PKG_CONFIG_PATH = "${pkgs.openssl.dev}/lib/pkgconfig";
+
+	# Add glibc, clang, glib, and other headers to bindgen search path
+	BINDGEN_EXTRA_CLANG_ARGS =
+	# Includes normal include path
+	(builtins.map (a: ''-I"${a}/include"'') [
+		# Dev Libraries
+		pkgs.glibc.dev
+		pkgs.openssl.dev
+	])
+	# Includes with special directory paths
+	++ [
+		''-I"${pkgs.llvmPackages_latest.libclang.lib}/lib/clang/${pkgs.llvmPackages_latest.libclang.version}/include"''
+		''-I"${pkgs.glib.dev}/include/glib-2.0"''
+		''-I${pkgs.glib.out}/lib/glib-2.0/include/''
+	];
```

Comment:
This is also only correct for Linux (glibc and glib). The [nixpkgs package](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/ka/kanidm/generic.nix) also doesn't have this at all, is this only recently necessary?

Reply:
can BSD even run nix. This shell.nix only makes sense within the nix context and afaik that focuses on linux.

Reply:
The big other platform nixpkgs supports (and probably the only one it's feasible to do Rust dev with Nix on right now tbh) which is pretty popular is macOS.

Reply:
Ah, I didn't even realize nix ran on macos. That is indeed a consideration. I feel like the main demographic is still linux so if we have the choice between it working and being cross platform I'll take the former. But if we can get it working without limiting it to only linux then all the better

Reply:
This has all been removed too